evennia/contrib/turnbattle/tb_magic.py

```diff
@@ -0,0 +1,1293 @@
+        longer needed once a fight ends.
+    """
+    for attr in character.attributes.all():
+        if attr.key[:7] == "combat_":  # If the attribute name starts with 'combat_'...
+            character.attributes.remove(key=attr.key)  # ...then delete it!
+
+
+def is_in_combat(character):
+    """
+    Returns true if the given character is in combat.
+
+    Args:
+        character (obj): Character to determine if is in combat or not
+
+    Returns:
+        (bool): True if in combat or False if not in combat
+    """
+    return bool(character.db.combat_turnhandler)
+
+
+def is_turn(character):
+    """
+    Returns true if it's currently the given character's turn in combat.
+
+    Args:
+        character (obj): Character to determine if it is their turn or not
+
+    Returns:
+        (bool): True if it is their turn or False otherwise
+    """
+    turnhandler = character.db.combat_turnhandler
+    currentchar = turnhandler.db.fighters[turnhandler.db.turn]
+    return bool(character == currentchar)
+
+
+def spend_action(character, actions, action_name=None):
+    """
+    Spends a character's available combat actions and checks for end of turn.
+
+    Args:
+        character (obj): Character spending the action
+        actions (int) or 'all': Number of actions to spend, or 'all' to spend all actions
+
+    Kwargs:
+        action_name (str or None): If a string is given, sets character's last action in
+        combat to provided string
+    """
+    if not is_in_combat(character):
+        return
+    if action_name:
+        character.db.combat_lastaction = action_name
+    if actions == 'all':  # If spending all actions
+        character.db.combat_actionsleft = 0  # Set actions to 0
+    else:
+        character.db.combat_actionsleft -= actions  # Use up actions.
+        if character.db.combat_actionsleft < 0:
+            character.db.combat_actionsleft = 0  # Can't have fewer than 0 actions
+    character.db.combat_turnhandler.turn_end_check(character)  # Signal potential end of turn.
+
+
+"""
+----------------------------------------------------------------------------
+CHARACTER TYPECLASS
+----------------------------------------------------------------------------
+"""
+
+
+class TBMagicCharacter(DefaultCharacter):
+    """
+    A character able to participate in turn-based combat. Has attributes for current
+    and maximum HP, and access to combat commands.
+    """
+
+    def at_object_creation(self):
+        """
+        Called once, when this object is first created. This is the
+        normal hook to overload for most object types.
+        """
+        self.db.max_hp = 100  # Set maximum HP to 100
+        self.db.hp = self.db.max_hp  # Set current HP to maximum
+        self.db.spells_known = [] # Set empty spells known list
+        self.db.max_mp = 20 # Set maximum MP to 20
+        self.db.mp = self.db.max_mp # Set current MP to maximum
+        """
```

Comment:
Please move this up with the normal docstring of this method. Use a `Notes:` section if you want to separate it from the header.